robot/impl/resource_manager.go

```diff
@@ -1023,14 +1030,59 @@ func (manager *resourceManager) markResourcesRemoved(
 func (manager *resourceManager) removeOrphanedResources(ctx context.Context,
 	rNames []resource.Name,
 ) {
-	// Ignore returned resources to close, as removeMarkedAndClose will handle.
 	manager.markResourcesRemoved(rNames, nil)
-	if _, err := manager.removeMarkedAndClose(ctx, nil); err != nil {
+	if err := manager.removeMarkedAndClose(ctx, nil); err != nil {
 		manager.logger.Errorw("error removing and closing marked resources",
 			"error", err)
 	}
 }
 
+// createConfig will create a config.Config based on the current state of the
+// resource graph, processManager and moduleManager.
+func (manager *resourceManager) createConfig() *config.Config {
+	conf := &config.Config{}
+
+	for _, resName := range manager.resources.Names() {
+		// Ignore non-local resources.
+		if resName.ContainsRemoteNames() {
+			continue
+		}
+		gNode, ok := manager.resources.Node(resName)
+		if !ok {
+			continue
+		}
+		resConf := gNode.Config()
+
+		// gocritic will complain that this if-else chain should be a switch, but
+		// it's really a mix of == and bool method checks.
+		//
+		//nolint: gocritic
+		if resName.API == client.RemoteAPI {
+			remoteConf, err := rutils.AssertType[*config.Remote](resConf.ConvertedAttributes)
+			if err != nil {
+				manager.logger.Errorw("remote has unexpected type for ConvertedAttributes",
```

Comment:
```suggestion
				manager.logger.Errorw("error getting remote config",
```

the error will look duplicative otherwise